src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -334,6 +346,36 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
                                                  local_service_account);
   grpc_gcp_AltsContext_set_peer_rpc_versions(
       context, const_cast<grpc_gcp_RpcProtocolVersions*>(peer_rpc_version));
+
+// START OF PLUMBING
+  grpc_gcp_HandshakerResult* ncresult = const_cast<grpc_gcp_HandshakerResult*>(hresult);
+  grpc_gcp_Identity* peer_identity = const_cast<grpc_gcp_Identity*>(identity);
+  if(peer_identity == nullptr ) { //|| *peer_attributes_counter == nullptr) {
+    gpr_log(GPR_ERROR, "Null Peer Identity.");
+    return TSI_FAILED_PRECONDITION;
+  }
+
+  if ( ! grpc_gcp_Identity_has_attributes(const_cast<grpc_gcp_Identity*>(peer_identity))) {
+        gpr_log(GPR_ERROR, "XOXO  Failed to DEFINE PEER ATTRIBUTES ENTRY AND VALUES");
+      }
+
+  size_t iter;
```

Comment:
I guess that `iter` needs to be initialized to `UPB_MAP_BEGIN` to begin with the first element of the map.

Reply:
Thanks for finding this! Switching to that allows me to pass all my tests now